ng/PowerShellScriptGenerator.ts

```diff
@@ -0,0 +1,475 @@
+﻿module armExplorer {
+
+    enum ResourceActions {
+        PATCH,
+        PUT,
+        GET,
+        CREATE
+    }
+
+    enum HttpVerb {
+        GET,
+        POST,
+        DELETE
+    }
+
+    function GETPSObjectFromJSON(json: string, nestingLevel: number): string {
+        var tabs = "";
+        for (var i = 0; i < nestingLevel; i++) {
+            tabs += "\t";
+        }
+        var jsonObj = JSON.parse(json);
+        if (typeof jsonObj === "string") {
+            return "\"" + jsonObj + "\"";
+        }
+        else if (typeof jsonObj === "boolean") {
+            return jsonObj.toString();
+        }
+        else if (typeof jsonObj === "number") {
+            return jsonObj.toString();
+        }
+        else if (Array.isArray(jsonObj)) {
+            var result = "(\n";
+            for (var i = 0; i < jsonObj.length; i++) {
+                result += tabs + "\t" + GETPSObjectFromJSON(JSON.stringify(jsonObj[i]), nestingLevel + 1) + "\n";
+            }
+            return result + tabs + ")";
+        }
+        else if (typeof jsonObj === "object") {
+            var result = "@{\n";
+            for (var prop in jsonObj) {
+                if (jsonObj.hasOwnProperty(prop)) {
+                    result += tabs + "\t" + prop + " = " + GETPSObjectFromJSON(JSON.stringify(jsonObj[prop]), nestingLevel + 1) + "\n";
+                }
+            }
+            return result + tabs + "}\n";
+        }
+        return json;
+    }
+
+    enum RMURLPARTS {
+        PROTOCOL,
+        BLANK,
+        DOMAINNAME,
+        SUBSCRIPTIONS_KEY,
+        SUBSCRIPTIONS_VALUE,
+        RESOURCEGROUPS_KEY,
+        RESOURCEGROUPS_VALUE, // Urls with length <= URLPARTS_RESOURCEGROUPS_VALUE will only include resourceId
+        PROVIDER_KEY,
+        PROVIDER_VALUE, // start of resourcetype followed by 9, 11, 13, ... ( 8,9 will always exists, 11,13.. is optional)
+        RESOURCETYPE1_NAME,
+        RESOURCETYPE1_VALUE // start of resourcename followed by 12, 14,...(12, 14,... is optional)
```

Comment:
i originally had them as individual constants and i usually have constants all caps for easy identification and just directly copied them into an enum. looks like the typescript coding guidelines also call for PascalCase like you mentioned. https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines